### ISSUE TITLE
How to write TensocrflowTFRecords at this format 

### ISSUE THREAD

**Issue by allansdefreitas** [keras-team/keras#16795]
Greetings!

I need to write Tensorflow TFRecords files in this specific format, for input to an object detection model:

```
PrefetchDataset element_spec={
    'image': TensorSpec(shape=(None, None, 3), dtype=tf.uint8, name=None), 
    'image/filename': TensorSpec(shape=(), dtype=tf.string, name=None), 
    'image/id': TensorSpec(shape=(), dtype=tf.int64, name=None), 
    'objects': {
        'area': TensorSpec(shape=(None,), dtype=tf.int64, name=None), 
        'bbox': TensorSpec(shape=(None, 4), dtype=tf.float32, name=None), 
        'id': TensorSpec(shape=(None,), dtype=tf.int64, name=None), 
        'is_crowd': TensorSpec(shape=(None,), dtype=tf.bool, name=None), 
        'label': TensorSpec(shape=(None,), dtype=tf.int64, name=None)}
}
```

I want to implement like this [implementation](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/keras_recipes/ipynb/creating_tfrecords.ipynb), but at above specified format (where ‘ojbects’ key have another subkeys in it). 

Thanks a lot!
Allan S.

**Comment by sushreebarsa** [comment]
@allansdefreitas I tried to replicate the issue reported, could you please have a look at this [gist](https://colab.research.google.com/gist/sushreebarsa/4f53b39276445bf93a83e9e33420fa80/creating_tfrecords.ipynb) and confirm the same?
Thank you!

**Comment by allansdefreitas** [comment]
> @allansdefreitas I tried to replicate the issue reported, could you please have a look at this [gist](https://colab.research.google.com/gist/sushreebarsa/4f53b39276445bf93a83e9e33420fa80/creating_tfrecords.ipynb) and confirm the same? Thank you!

Yes, @sushreebarsa, I want to have TFRecords like:
bbox: [473.07 395.93  38.65  28.67]
area: 702.1057739257812
category_id: 18
id: 1768
image_id: 289343
path: b'datasets/val2017/000000289343.jpg'
Image shape: (640, 529, 3)

But in this format:

    'image': TensorSpec(shape=(None, None, 3), dtype=tf.uint8, name=None), 
    'image/filename': TensorSpec(shape=(), dtype=tf.string, name=None), 
    'image/id': TensorSpec(shape=(), dtype=tf.int64, name=None), 
    'objects': {
        'area': TensorSpec(shape=(None,), dtype=tf.int64, name=None), 
        'bbox': TensorSpec(shape=(None, 4), dtype=tf.float32, name=None), 
        'id': TensorSpec(shape=(None,), dtype=tf.int64, name=None), 
        'is_crowd': TensorSpec(shape=(None,), dtype=tf.bool, name=None), 
        'label': TensorSpec(shape=(None,), dtype=tf.int64, name=None)}

**Comment by gadagashwini** [comment]
Hi @allansdefreitas, You can customize the Tf TfRecords file format based on the requirements.
Follow the example given on the Tensorflow site [here](https://www.tensorflow.org/tutorials/load_data/tfrecord#write_the_tfrecord_file). Thank you!
